### PR TITLE
Add localization support

### PR DESCRIPTION
**Pull request by LinuxSuRen**

### CODE REVIEW

**Comment by LinuxSuRen** [comment]
I did't get your point. You mean compare with other branch-source plugins? Like gitlab or bitbucket?

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java [hunk 1]
```diff
@@ -1427,9 +1427,9 @@ public boolean test(SCMTraitDescriptor<?> scmTraitDescriptor) {
                         }
                     },
                     true, result);
-            NamedArrayList.select(all, "Within repository", NamedArrayList.anyOf(NamedArrayList.withAnnotation(Discovery.class),NamedArrayList.withAnnotation(Selection.class)),
+            NamedArrayList.select(all, Messages.within_repository(), NamedArrayList.anyOf(NamedArrayList.withAnnotation(Discovery.class),NamedArrayList.withAnnotation(Selection.class)),
```
Comment (bitwiseman) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]: GitHubSCMNavigator.
```suggestion
            NamedArrayList.select(all, Messages.GitHubSCMNavigator_withinRepository(), NamedArrayList.anyOf(NamedArrayList.withAnnotation(Discovery.class),NamedArrayList.withAnnotation(Selection.class)),
```

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java [hunk 2]
```diff
@@ -1427,9 +1427,9 @@ public boolean test(SCMTraitDescriptor<?> scmTraitDescriptor) {
...
                     true, result);
-            NamedArrayList.select(all, "General", null, true, result);
+            NamedArrayList.select(all, Messages.general(), null, true, result);
```
Comment (bitwiseman) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java]:
```suggestion
            NamedArrayList.select(all, Messages.GitHubSCMNavigator_general(), null, true, result);
```

**Comment by bitwiseman** [comment]
No, I mean the same naming is other messages in this plugin.

File: src/main/resources/org/jenkinsci/plugins/github_branch_source/Messages.properties [hunk 3]
```diff
@@ -56,3 +58,6 @@ SSHCheckoutTrait.missingCredentials=The currently configured credentials cannot
 SSHCheckoutTrait.useAgentKey=- use build agent''s key -
 TagDiscoveryTrait.authorityDisplayName=Trust origin tags
 TagDiscoveryTrait.displayName=Discover tags
+
+general=General
```
Comment (bitwiseman) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/Messages.properties]:
```suggestion
GitHubSCMNavigator.general =General
```

File: src/main/resources/org/jenkinsci/plugins/github_branch_source/Messages.properties [hunk 4]
```diff
@@ -56,3 +58,6 @@ SSHCheckoutTrait.missingCredentials=The currently configured credentials cannot
...
+within.repository=Within repository
```
Comment (bitwiseman) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/Messages.properties]:
```suggestion
GitHubSCMNavigator.withinRepository=Within repository
```

File: src/main/java/org/jenkinsci/plugins/github_branch_source/OriginPullRequestDiscoveryTrait.java [hunk 5]
```diff
@@ -136,7 +136,7 @@ public boolean includeCategory(@NonNull SCMHeadCategory category) {
          */
         @Override
         public String getDisplayName() {
-            return "Discover pull requests from origin";
+            return Messages.discover_pull_requests_from_origin();
```
Comment (bitwiseman) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/OriginPullRequestDiscoveryTrait.java]:
```suggestion
            return Messages.OriginPullRequestDiscoveryTrait_discoverPullRequestsFromOrigin();
```

File: src/main/resources/org/jenkinsci/plugins/github_branch_source/Messages.properties [hunk 6]
```diff
@@ -43,6 +43,8 @@ GitHubSCMSource.did_you_mean_to_use_to_match_all_branches=Did you mean to use *
 GitHubSCMSource.NoMatchingOwner=Could not find owner: {0}
 GitHubSCMSource.CouldNotConnectionGithub=Could not connect github with credential: {0}
 
+discover.pull.requests.from.origin=Discover pull requests from origin
```
Comment (bitwiseman) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/Messages.properties]:
```suggestion
OriginPullRequestDiscoveryTrait.discoverPullRequestsFromOrigin=Discover pull requests from origin
```

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java [hunk 7]
```diff
@@ -2284,11 +2284,11 @@ private static ListBoxModel nameAndValueModel(Collection<String> items) {
                 }
             }
             List<NamedArrayList<? extends SCMTraitDescriptor<?>>> result = new ArrayList<>();
-            NamedArrayList.select(all, "Within repository", NamedArrayList
+            NamedArrayList.select(all, Messages.within_repository(), NamedArrayList
```
Comment (bitwiseman) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java]:
```suggestion
            NamedArrayList.select(all, Messages.GitHubSCMNavigator_withinRepository(), NamedArrayList
```

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java [hunk 8]
```diff
@@ -2284,11 +2284,11 @@ private static ListBoxModel nameAndValueModel(Collection<String> items) {
...
                             .anyOf(NamedArrayList.withAnnotation(Discovery.class),
                                     NamedArrayList.withAnnotation(Selection.class)),
                     true, result);
-            NamedArrayList.select(all, "General", null, true, result);
+            NamedArrayList.select(all, Messages.general(), null, true, result);
```
Comment (bitwiseman) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java]:
```suggestion
            NamedArrayList.select(all, Messages. GitHubSCMNavigator_general(), null, true, result);
```

**Comment by LinuxSuRen** [comment]
@bitwiseman Sorry for my delay.

**Pull request merged**